compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -1704,6 +1704,85 @@ unsupported {} from `{}` with element `{}` of size `{}` to `{}`"#,
     bitwise_red!(simd_reduce_all: vector_reduce_and, true);
     bitwise_red!(simd_reduce_any: vector_reduce_or, true);
 
+    if name == sym::simd_cast_ptr {
+        require_simd!(ret_ty, "return");
+        let (out_len, out_elem) = ret_ty.simd_size_and_type(bx.tcx());
+        require!(
+            in_len == out_len,
+            "expected return type with length {} (same as input type `{}`), \
+                  found `{}` with length {}",
+            in_len,
+            in_ty,
+            ret_ty,
+            out_len
+        );
+
+        match in_elem.kind() {
+            ty::RawPtr(_) => {}
```

Comment:
Is it worth checking that this is a ptr to a sized type, or more genreally not a wide ptr?

Reply:
I suppose that should really be rejected by `repr(simd)` (I'm not sure if it is)

Reply:
Also, how are wide casts handled by the backend, I wonder? Like `*const [T] as *const [U]`?  If we ever had vectors of fat pointers, I suppose that would still be something valid.

Reply:
The backend code for casts has tons of special cases to cover all the possibilities.

Reply:
Just confirmed, `repr(simd)` does already reject wide pointers: https://github.com/rust-lang/rust/blob/ffa77332c630b43cb92f6256b8e404198820fa03/compiler/rustc_middle/src/ty/layout.rs#L909-L919

Reply:
Okay so you don't think hat is worth double-checking here? (I usually prefer not to rely on checks that are done in other crates.)

Reply:
I'm a little hesitant since checking if it's sized here seems to require the parameter environment etc.  Either way, if `repr(simd)` is changed to allow wide pointers, `simd_arith_offset` needs to be updated as well.

Reply:
I read through some of the docs and this was much more of a non-issue than I thought, so I added in a check to ensure the pointed-to types are sized when casting.